File: src/items/Item.py
'''
This class will be the one that the other items will adquire.

It will have the atributes about the item (Name, sell-in, quality).
'''

class Item:

    # Constructor
    def __init__(self, name, sell_in, quality):
        
        self.name = name
        self.sell_in = sell_in
        self.quality = quality

    # Changing the built-in method to call this object in case of needed.
    def __repr__(self):
        
        return f"{self.name} {self.sell_in} {self.quality}"

    # Getters/Setters
    def getName(self):

        return self.name

    def getSellIn(self):

        return self.name

    def getQuality(self):

        return self.name

    def setName(self, new_name):

        self.name = new_name

    def setSellIn(self, new_sell_in):

        self.name = new_sell_in

    def setQuality(self, new_quality):

        self.name = new_quality

File: java/domain/Item.java
package java.domain;

/**
 * This java class will be the one that NormalItem will inerhitance and the
 * future methods and atributes for each item.
 * 
 * This will be the class that will construct the others, so we will set it
 * as final, so it cannot be changed in the future.
 */

final class Item{
    
    String name;
    int sellIn;
    int quality;

    // Constructor to the method
    public Item(String name, int sellIn, int quality){

        this.name = name;
        this.sellIn = sellIn;
        this.quality = quality;
    }

    @Override
    public String toString(){

        return ("Item name: "+ this.name +
                ". Sell-In date: "+ this.sellIn + 
                ". Actual quality: " + this.quality + ". ");

    }

    // For being strict to the initial kata, i will comment this part that
    // I created to keep in mind encapsulation.


    // Getters and Setters
    // public String getName(){
        
    //     return this.name;
    // }

    // public int getSellIn(){

    //     return this.sellIn;
    // }

    // public int getQuality(){

    //     return this.quality;
    // }

    // public void setName(String newName){

    //     this.name = newName;
    // }

    // public void setSellIn(int value){

    //     this.sellIn = value;
    // }

    // public void setQuality(int value){

    //     this.quality = value; 
    // }

    // Override the method toString, to all the classes.

}
